src/CollapsibleField.js

```diff
@@ -241,35 +292,59 @@ class CollapsibleField extends Component {
 
   render() {
     let {
-      schema: { title },
+      schema: { title, properties = {} },
       uiSchema,
       registry: { fields },
       idSchema: { $id } = {},
       name,
-      formContext
+      formContext,
+      formData,
+      onChange: propsOnChange
     } = this.props;
     let { collapsed, AddElement } = this.state;
-    let { collapse: { field } } = uiSchema;
+    let { collapse: { collapsibleHeaderElements = {}, field } } = uiSchema;
     let CollapseElement = fields[field];
+    let { elements = [] } = collapsibleHeaderElements;
     // uischema retains the value form the state
     uiSchema.collapse.collapsed = this.state.collapsed;
 
     title = uiSchema["ui:title"] ? uiSchema["ui:title"] : title ? title : name;
     let customizedId = collapsed ? $id : undefined;
+
+    //remove header elements from the schema
+    let headerElementsSchemas = {};
+    let propertiesNoHeader = { ...properties };
+    elements.forEach(key => {
+      if (propertiesNoHeader[key]) {
+        headerElementsSchemas[key] = propertiesNoHeader[key];
+        delete propertiesNoHeader[key];
+      }
+    });
+
+    const collapseElementProps = {
+      ...this.props,
+      schema: { ...this.props.schema, properties: propertiesNoHeader }
```

Comment:
hmm, we should probably do the same for ui schema too and readjust the ui-order as well.

Reply:
I think i took care of ui-order in the schema-generator diagnosisv2